src/Analyzers/Core/CodeFixes/AddParameter/AddParameterService.cs

```diff
@@ -130,18 +136,48 @@ public static async Task<Solution> AddParameterAsync(
                 }
 
                 if (method.MethodKind == MethodKind.ReducedExtension && insertionIndex < existingParameters.Count)
-                {
                     insertionIndex++;
+
+                var memberToAssignTo = await GetMemberToAssignToAsync(semanticDocument).ConfigureAwait(false);
+                if (memberToAssignTo != null)
+                {
+                    var initializeParameterService = documentLookup.Key.GetRequiredLanguageService<IInitializeParameterService>();
+                    var body = initializeParameterService.GetBody(methodNode);
+                    if (semanticDocument.SemanticModel.GetOperation(body, cancellationToken) is IBlockOperation blockOperation)
+                    {
+                        initializeParameterService.AddAssignment(
+                            methodNode, blockOperation, parameterSymbol, memberToAssignTo, editor);
```

Comment:
and, if so, calls into the InitializeParameterService to actually add the assignment.

By reusing these services, we share as much code logic as possible, so the features don't have to dupe them.